### PR TITLE
emx2 pyclient skeleton

### PR DESCRIPTION
**Pull request by erikzwart**
emx2 pyclient skeleton

todo:
- [x] remove what is not essential
- [x] add to build pipeline
- [ ] add versioning
- [ ] previews should go to [pypi test](https://test.pypi.org/project/molgenis-emx2-client/)
- [ ] release should go to pypi
- [x] proof that we can install the packages from pypi test

### CODE REVIEW

File: tools/pyclient/.env_example [hunk 1]
```diff
@@ -0,0 +1,4 @@
+MG_URL = https://MOLGENIS.molgeniscloud.org
```
Comment (connoratrug) [inline tools/pyclient/.env_example:1]: suggest to use https://emx2.dev.molgenis.org/ and pet store as example ?

File: tools/pyclient/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+pandas~=1.5.3
```
Comment (connoratrug) [inline tools/pyclient/requirements.txt:1]: does the client need pandas ?

File: tools/pyclient/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+# Configuration
+Create file `.env` using  `.env_example` as a reference.
+
+
+
+# How to use
+After installing the dependencies and creating the `.env` file run
+    
+    python -m molgenis
```
Comment (connoratrug) [inline tools/pyclient/README.md:45]: suggest to replace with build step en use pip in how to use sample

File: tools/pyclient/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
+# Installation
+
```
Comment (connoratrug) [inline tools/pyclient/README.md:2]: installation / usage should be updated to something like ```pip install ....``` . And then we can have a separate section on development

File: tools/pyclient/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#
+# This file is autogenerated by pip-compile with Python 3.11
```
Comment (connoratrug) [inline tools/pyclient/requirements.txt:2]: if it is generated it probably should be added to git ignore

File: tools/pyclient/src/molgenis/__init__.py [hunk 6]
```diff
@@ -0,0 +1,10 @@
+from .client import Client
+
+from importlib import resources
+try:
+    import tomllib
+except ModuleNotFoundError:
+    import tomli as tomllib
+
+# Version of the realpython-reader package
```
Comment (connoratrug) [inline tools/pyclient/src/molgenis/__init__.py:9]: remove comment and example code

File: tools/pyclient/src/molgenis/__main__.py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+    except KeyError:
+        log.error('Make sure you filled in all variables in the .env file, script will exit now.')
+        sys.exit()
+
+    molgenis_client = Client(
+        url=url,
+        database=database,
+        email=username,
+        password=password
```
Comment (connoratrug) [inline tools/pyclient/src/molgenis/__main__.py:35]: maybe remove from Client init ?

File: tools/pyclient/src/molgenis/client.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    """
+    def __init__(self, url: str, database: str, email: str, password: str) -> None:
+        self.url = url
+        self.database = database
+
+        self.session = requests.Session()
+        self.graphqlEndpoint = f'{self.url}/{self.database}/graphql'
+        self.apiEndpoint = f'{self.url}/{self.database}/api'
+
+        self.signin(email, password)
```
Comment (connoratrug) [inline tools/pyclient/src/molgenis/client.py:20]: suggest to remove

File: tools/pyclient/README.md [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    git clone git@github.com:molgenis/molgenis-emx2.git
+
+Change the working directory to `.../tools/pyclient`
+
+### Create a virtual Python environment
+
+On macOS:
+
+    python -m venv venv
```
Comment (mswertz) [inline tools/pyclient/README.md:28]: I think we should use python3?

**Review by mswertz — COMMENTED** [review]
until now I assumed you want to use python3. Now, we can of course link python to python3 on our build server? Please advice.

File: tools/pyclient/LICENSE [hunk 10]
```diff
@@ -0,0 +1,165 @@
+GNU LESSER GENERAL PUBLIC LICENSE
```
Comment (mswertz) [inline tools/pyclient/LICENSE:1]: can we somehow symlink to the overal license?

**Review by mswertz — APPROVED** [review]

File: tools/pyclient/LICENSE (hunk 10 above)
Comment (erikzwart) [reply tools/pyclient/LICENSE]: added symlink to LICENSE

**Pull request merged**